src/shell/focus/mod.rs

```diff
@@ -219,9 +223,28 @@ fn update_focus_state(
     // update keyboard focus
     if let Some(keyboard) = seat.get_keyboard() {
         if check_update_cursor && state.common.config.cosmic_conf.cursor_follows_focus {
-            let last_active = ActiveFocus::get(seat);
             if ActiveFocus::get(seat).as_ref() != target {
-                todo!("update cursor position");
+                // get the top left corner of the target element
+                let element = state
+                    .common
+                    .shell
+                    .read()
+                    .unwrap()
+                    .focused_element(target.unwrap())
+                    .unwrap();
+                let top_left = element.active_window_geometry().loc.to_f64();
```

Comment:
This is probably not in global coordinates, which would be what you need.
You need to use something similar to [this](https://github.com/pop-os/cosmic-comp/blob/master/src/shell/mod.rs#L2740-L2760) for that (perhabs that should be a new `Shell::element_geometry` method).

Reply:
if I just need the location as global coordinates, would calling `element.active_window_geometry.as_global()` work? Also if I need globabl coordinates, I'm guessing `PointerFocusTarget.motion` is not the function I need to call given MotionEvent want a logical coordinate. What should I use instead?